packages/session-replay-browser/src/session-replay.ts

```diff
@@ -193,9 +196,8 @@ export class SessionReplay implements AmplitudeSessionReplay {
   focusListener = () => {
     // Restart recording on focus to ensure that when user
     // switches tabs, we take a full snapshot
-    if (this.sessionTargetingMatch) {
-      this.recordEvents();
-    }
+    this.stopRecordingEvents();
+    this.captureEventsIfShould();
```

Comment:
Depending on the site, this may cause a lot of full snapshots. For example try Google Search and click on the nine dots next to your profile picture in the top right. For me that causes two focus events without switching tabs. I wonder if there is a better way to support tab switching.

Reply:
Can we wrap this for targeting changes specifically or move to another PR? I'm worried about the potential performance impact here.

Reply:
This is actually how it already works! So this is preserving current functionality here: https://github.com/amplitude/Amplitude-TypeScript/blob/v1.x/packages/session-replay-browser/src/session-replay.ts#L188

And this has been how it's worked for a long while - we used to explicitly stop recording on blur, and restart on focus, I recently changed it to be stop and restart on focus only.

The only change happening in this PR is that this is more explicit than it used to be, that implicit behavior was happening within `recordEvents`. I need to make this change in this PR to allow `evaluateTargetingAndCapture` to call `recordEvents` _without_ causing a full dom snapshot, and without restarting recording.

Maybe an overexplanatory loom here: https://www.loom.com/share/57e2abca82714b85bf1f4b64b52309b4

Reply:
Gotcha. Thanks for the clarification!